ballerina/service.bal

```diff
@@ -15,54 +15,66 @@
+        isolated resource function get dryRun(graphql:Context context, graphql:Field 'field, SubgraphInput schema, boolean isForced = false) returns CompositionResult|error? {
+            registry:CompositionResult|parser:SchemaError[]|merger:MergeError[]|registry:OperationCheckError[] result = check self.registry.dryRun(schema, isForced);
+            if result is registry:CompositionResult {
+                return new CompositionResult(result);
+            } else {
+                check returnErrors(context, 'field, result);
+                return;
+            }
         }
-    }
 
-    isolated resource function get subgraph(string name) returns Subgraph|error {
-        return new Subgraph(check self.registry.getSubgraphByName(name));
-    }
+        isolated resource function get subgraph(string name) returns Subgraph|error {
+            return new Subgraph(check self.registry.getSubgraphByName(name));
+        }
 
-    isolated resource function get diff(graphql:Context context, graphql:Field 'field, string newVersion, string oldVersion) returns differ:SchemaDiff[]|error {
-        return check self.registry.getDiff(newVersion, oldVersion);
-    }
+        isolated resource function get diff(graphql:Context context, graphql:Field 'field, string newVersion, string oldVersion) returns differ:SchemaDiff[]|error {
+            return check self.registry.getDiff(newVersion, oldVersion);
+        }
 
-    isolated remote function publishSubgraph(graphql:Context context, graphql:Field 'field, SubgraphInput schema, boolean isForced = false) returns CompositionResult|error? {
-        registry:CompositionResult|parser:SchemaError[]|merger:MergeError[]|registry:OperationCheckError[] result = check self.registry.publishSubgraph(schema, isForced);
-        if result is registry:CompositionResult {
-            return new CompositionResult(result);
-        } else {
-            check returnErrors(context, 'field, result);
-            return;
+        isolated remote function publishSubgraph(graphql:Context context, graphql:Field 'field, SubgraphInput schema, boolean isForced = false) returns CompositionResult|error? {
+            registry:CompositionResult|parser:SchemaError[]|merger:MergeError[]|registry:OperationCheckError[] result = check self.registry.publishSubgraph(schema, isForced);
+            if result is registry:CompositionResult {
+                return new CompositionResult(result);
+            } else {
+                check returnErrors(context, 'field, result);
+                return;
+            }
         }
-    }
+    };
+    return schemaRegistryService;
+}
+
+public function startRegistry(datasource:Datasource datasource, int port) returns error? {
+    graphql:Listener graphqlListener = check new (port);
+    graphql:Service registryService = getSchemaRegistryService(datasource);
+    check graphqlListener.attach(registryService);
+    check graphqlListener.'start();
+    runtime:registerListener(graphqlListener);
```

Comment:
Shouldn't we register the listener first?

```suggestion
    runtime:registerListener(graphqlListener);
    check graphqlListener.attach(registryService);
    check graphqlListener.'start();
```

Reply:
I had the same problem, but I decided to follow these ([dynamic http listener](https://ballerina.io/learn/by-example/dynamic-listener/), [dynamic graphql listener test](https://github.com/ballerina-platform/module-ballerina-graphql/blob/63450ac938c84fb98338a6a584b87f030c01bce0/compiler-plugin-tests/src/test/resources/ballerina_sources/validator_tests/23_dynamically_attaching_service/service.bal#L39)) references. What should we do?

Reply:
Hmm, right. That makes sense. Let's keep it as it is.